src/setup/txt_joybinput.c

```diff
@@ -381,6 +574,93 @@ static void TXT_JoystickInputDrawer(TXT_UNCAST_ARG(joystick_input))
     }
 }
 
+static int GetGamepadButtonIndex(int button)
+{
+    int i;
+
+    for (i = 0; i < arrlen(gamepad_buttons); ++i)
+    {
+        if (button == gamepad_buttons[i])
+        {
+            return i;
+        }
+    }
+
+    I_Error("Couldn't find button index.");
+    return -1;
+}
+
+static void GetGamepadButtonDescription(int vbutton, char *buf, size_t buf_len)
+{
+    int index;
+
+    index = GetGamepadButtonIndex(PhysicalForVirtualButton(vbutton));
+
+    if (index < 0)
+    {
+        return;
+    }
+
+    switch (gamepad_type)
+    {
+        case SDL_CONTROLLER_TYPE_XBOX360:
+            M_snprintf(buf, buf_len, "%s", xbox360_buttons[index]);
+            break;
+
+        case SDL_CONTROLLER_TYPE_XBOXONE:
+            M_snprintf(buf, buf_len, "%s", xboxone_buttons[index]);
+            break;
+
+        case SDL_CONTROLLER_TYPE_PS3:
+            M_snprintf(buf, buf_len, "%s", ps3_buttons[index]);
+            break;
+
+        case SDL_CONTROLLER_TYPE_PS4:
+            M_snprintf(buf, buf_len, "%s", ps4_buttons[index]);
+            break;
+
+        case SDL_CONTROLLER_TYPE_PS5:
+            M_snprintf(buf, buf_len, "%s", ps5_buttons[index]);
+            break;
+
+        case SDL_CONTROLLER_TYPE_NINTENDO_SWITCH_PRO:
+            M_snprintf(buf, buf_len, "%s", switchpro_buttons[index]);
+            break;
+
+        default:
+            M_snprintf(buf, buf_len, "BUTTON #%i",
+                       PhysicalForVirtualButton(vbutton) + 1);
+            break;
+    }
+}
+
+static void TXT_GamepadInputDrawer(TXT_UNCAST_ARG(joystick_input))
+{
+    TXT_CAST_ARG(txt_joystick_input_t, joystick_input);
+    char buf[20];
```

Comment:
Add a comment about the size of this buffer. What needs to fit into it. No need for symbolic constant since it's only in this one place.